_source/_posts/2024-08-15-otp-over-sms.md

```diff
@@ -0,0 +1,178 @@
+tags: [devops]
+tweets:
+- ""
+image: blog/telephony/social.jpg
+type: conversion
+---
+{% include toc.md %}
+
+## Approaches to keep sending OTP over SMS... for now
+
+"SMS has long played an important role as a universally applicable method of verifying a user's identity via one-time passcodes. And over the last decade, SMS and voice-based Multifactor Authentication has prevented untold attempts to compromise user accounts.
+
+But it's time to move on."
+
+– Ben King, VP Customer Trust: [BYO Telephony and the future of SMS at Okta](https://sec.okta.com/articles/2023/08/byo-telephony-and-future-sms-okta?_gl=1*50v0v8*_gcl_au*NzM2MTA4NjA5LjE3MTk5NTY2MDU.*_ga*MTE1NjAwNzQxNC4xNjY0OTc5MTk3*_ga_QKMSDV5369*MTcyMjg4MjA0Ni4yODIuMS4xNzIyODgzNDM3LjYwLjAuMA..&_ga=2.145329393.498111759.1722873073-1156007414.1664979197)
+
+## SMS/Voice is too SIMple
+
+The one-time passcode (OTP) you send using SMS or Voice may not go to the phone you want. SIM swapping–stealing someone else's phone number–lets bad actors receive the message or call with the code. They're one step closer to breaking into your system. And if all it takes is an account name and OTP, they may succeed. And it's not just SIM hacking; other issues include:
+
+* No phishing resistance
+
+* No control of the channel for sending secrets
+
+* No way to link a user to their device
+
+* Longer login times than other methods
+
+Okta [recommended moving away](https://www.okta.com/blog/2020/05/why-you-should-ditch-sms-as-an-auth-factor/)[ from SMS/Voice authentication](https://www.okta.com/blog/2020/05/why-you-should-ditch-sms-as-an-auth-factor/) some time ago. There are many other factors you can use for authentication, including:
+
+* Generating codes in an authenticator app such as Okta Verify, Authy, Google Authenticator, or 1Password.
+
+* FIDO2.0 (WebAuthn) which, in addition to phones, can use hardware keys and on-device authenticators.
+
+Soon, [Okta will](https://support.okta.com/help/s/article/bring-your-own-telephony-required-for-sms-and-voice?language=en_US)[ require you to bring your own telephony provider](https://support.okta.com/help/s/article/bring-your-own-telephony-required-for-sms-and-voice?language=en_US) to keep sending those codes. If you need time to move to a different method of verifying identity, you must configure your own provider for SMS/Voice.
+
+## Hooked on telephony
+
+You can send the OTP in the SMS/Voice flow using the [telephony inline hook]([https://help.okta.com/oie/en-us/content/topics/telephony/telephony-inline-hook.htm](https://help.okta.com/oie/en-us/content/topics/telephony/telephony-inline-hook.htm)). Okta uses the code or URL in the hook to send the OTP, though, as you'll see, the hook may not be called every time (and that's a good thing). When your hook fails to send the message or takes too long to update the status, Okta takes over sending the message. However, the number of those messages is heavily rate-limited.
```

Comment:
does a link to the page suffice? (meaning does it also need the entire URL written as well?)